### DISCUSSION TITLE
Footnotes not rendering as expected with MDX in Next.js

### DISCUSSION THREAD

**Discussion by mirhamasala** [remarkjs/remark#1058] [Q&A]
I have the following Markdown in a `.mdx` file

```
Most companies try to stay ahead of the curve when it comes to visual design, but for Planetaria we needed to create a brand that would still inspire us 100 years from now when humanity has spread across our entire solar system. [^1]

[^1]: Note 1
```

The HTML rendered is:

```
<p>Most companies try to stay ahead of the curve when it comes to visual design, but for Planetaria we needed to create a brand that would still inspire us 100 years from now when humanity has spread across our entire solar system<sup><a href="/posts/crafting-a-design-system-for-a-multiplanetary-future#user-content-fn-1">1</a></sup>.</p>

<section data-footnotes="true" class="footnotes">
  <h4 class="sr-only" id="footnote-label">Notes</h4>
  <ol>
    <li id="user-content-fn-1">
    <p>note 2 <a href="/posts/crafting-a-design-system-for-a-multiplanetary-future#user-content-fnref-1">↩</a></p>
  </li>
</ol>
</section>
```

As you can see, it doesn't add any of the attributes as in the example: `<p>Hallo<sup><a href="#user-content-fn-1" id="user-content-fnref-1" data-footnote-ref aria-describedby="footnote-label">1</a></sup></p>`

I'm not sure where the problem is.

Here's my `next.config.mjs`:

```
import nextMDX from '@next/mdx'
import remarkGfm from 'remark-gfm'
import remarkRehype from 'remark-rehype'
import rehypePrism from '@mapbox/rehype-prism'

/** @type {import('next').NextConfig} */
const nextConfig = {
  pageExtensions: ['jsx', 'mdx'],
  reactStrictMode: true,
  swcMinify: true,
  experimental: {
    newNextLinkBehavior: true,
    scrollRestoration: true,
  },
}

const withMDX = nextMDX({
  extension: /\.mdx?$/,
  options: {
    remarkPlugins: [
      remarkGfm,
      [remarkRehype, { footnoteLabel: 'Notes', footnoteLabelTagName: 'h4' }],
    ],
    rehypePlugins: [rehypePrism],
    providerImportSource: '@mdx-js/react',
  },
})

export default withMDX(nextConfig)
```

**Comment by wooorm** [comment]
I think the problem is that you are specifying `remarkRehype`.

MDX already specifies that. Using both likely breaks things terribly.

Remove the plugin, and pass the options as [`remarkRehypeOptions`](https://mdxjs.com/packages/mdx/#optionsremarkrehypeoptions).

**Reply by mirhamasala** [reply]
Changed. Thank you, @wooorm. Unfortunately, it didn't fix the problem.

```
const withMDX = nextMDX({
  extension: /\.mdx?$/,
  options: {
    remarkPlugins: [remarkGfm],
    rehypePlugins: [rehypePrism],
    remarkRehypeOptions: {
      footnoteLabel: 'Notes',
      footnoteLabelTagName: 'h2',
    },
    providerImportSource: '@mdx-js/react',
  },
})
```

**Reply by wooorm** [reply]
> it didn't fix the problem.

This is too vague. What is the problem?

**Reply by mirhamasala** [reply]
The back link doesn't work because no attributes are added to the section where it should link to (no `id`, `data-footnote-ref` nor `aria-describedby`. This is what the HTML looks like:

```
<p>Most companies try to stay ahead of the curve when it comes to visual design, but for Planetaria we needed to create a brand that would still inspire us 100 years from now when humanity has spread across our entire solar system<sup><a href="/posts/crafting-a-design-system-for-a-multiplanetary-future#user-content-fn-1">1</a></sup>.</p>
```

Instead of something like this:

`<p>Hallo<sup><a href="#user-content-fn-1" id="user-content-fnref-1" data-footnote-ref aria-describedby="footnote-label">1</a></sup></p>`

**Reply by wooorm** [reply]
My guess is that you are using old versions of packages. Make sure particularly `@mdx-js/loader`, and every remark and rehype plugin, is updated to the latest.
If it doesn’t work yet, make a small reproduction. 
See https://mdxjs.com/community/support/ for more info on Support.

**Reply by mirhamasala** [reply]
Thank you for your time and help @wooorm. I figured it out. I was using a custom `Link` [component](https://mdxjs.com/table-of-components/). It works without that. 🙏🏼

**Reply by wooorm** [reply]
Cool, glad it works Mirha!
